src/api/cvc4cpp.cpp

```diff
@@ -1873,348 +1906,868 @@ Term Solver::mkFalse(void) const { return d_exprMgr->mkConst<bool>(false); }
+        << "ABSTRACT_VALUE or CONST_RATIONAL or CONST_STRING";
+    Term res;
+    if (kind == ABSTRACT_VALUE)
+    {
+      res = d_exprMgr->mkConst(CVC4::AbstractValue(Integer(arg, 10)));
+      // do not call getType(), for abstract values, type can not be computed
+      // until it is substituted away
+    }
+    else if (kind == CONST_RATIONAL)
+    {
+      res = mkReal(arg);
+      (void)res.d_expr->getType(true);  /* kick off type checking */
+    }
+    else
+    {
+      res = d_exprMgr->mkConst(CVC4::String(arg));
+      (void)res.d_expr->getType(true);  /* kick off type checking */
+    }
+    return res;
+  }
+  catch (std::invalid_argument& e)
+  {
+    throw CVC4ApiException(e.what());
+  }
+  catch (TypeCheckingException& e)
+  {
+    throw CVC4ApiException(e.getMessage());
+  }
+}
+
+Term Solver::mkConst(Kind kind, const char* arg1, uint32_t arg2) const
+{
+  try
+  {
+    CVC4_API_ARG_CHECK_NOT_NULL(arg1);
+    CVC4_API_ARG_CHECK_EXPECTED(!std::string(arg1).empty(), arg1)
+        << "a non-empty string";
+    CVC4_API_KIND_CHECK_EXPECTED(kind == CONST_BITVECTOR, kind)
+        << "CONST_BITVECTOR";
+    Term res = mkBitVector(arg1, arg2);
+    (void)res.d_expr->getType(true);  /* kick off type checking */
+    return res;
+  }
+  catch (TypeCheckingException& e)
+  {
+    throw CVC4ApiException(e.getMessage());
+  }
+}
+
+Term Solver::mkConst(Kind kind, const std::string& arg1, uint32_t arg2) const
+{
+  try
+  {
+    CVC4_API_ARG_CHECK_EXPECTED(!arg1.empty(), arg1) << "a non-empty string";
+    CVC4_API_KIND_CHECK_EXPECTED(kind == CONST_BITVECTOR, kind)
+        << "CONST_BITVECTOR";
+    Term res = mkBitVector(arg1, arg2);
+    (void)res.d_expr->getType(true);  /* kick off type checking */
+    return res;
+  }
+  catch (TypeCheckingException& e)
+  {
+    throw CVC4ApiException(e.getMessage());
+  }
+}
+
+Term Solver::mkConst(Kind kind, uint32_t arg) const
+{
+  try
+  {
+    CVC4_API_KIND_CHECK_EXPECTED(kind == ABSTRACT_VALUE
+                                     || kind == CONST_RATIONAL
+                                     || kind == CONST_BITVECTOR,
+                                 kind)
+        << "ABSTRACT_VALUE or CONST_RATIONAL or CONST_BITVECTOR";
+    Term res;
+    if (kind == ABSTRACT_VALUE)
+    {
+      res = d_exprMgr->mkConst(CVC4::AbstractValue(Integer(arg)));
+      // do not call getType(), for abstract values, type can not be computed
+      // until it is substituted away
+    }
+    else if (kind == CONST_RATIONAL)
+    {
+      res = d_exprMgr->mkConst(CVC4::Rational(arg));
+      (void)res.d_expr->getType(true);  /* kick off type checking */
+    }
+    else
+    {
+      res = d_exprMgr->mkConst(CVC4::BitVector(arg));
+      (void)res.d_expr->getType(true);  /* kick off type checking */
+    }
+    return res;
+  }
+  catch (TypeCheckingException& e)
+  {
+    throw CVC4ApiException(e.getMessage());
+  }
+}
+
+Term Solver::mkConst(Kind kind, int32_t arg) const
```

Comment:
Templated helper for `int32_t`, `int64_t`, and `uint64_t`?

Reply:
Solved it differently, have a look.